internal/app/machined/pkg/system/services/etcd.go

```diff
@@ -409,6 +409,7 @@ func (e *Etcd) argsForInit(ctx context.Context, r runtime.Runtime, spec *etcdres
 		"peer-trusted-ca-file":               constants.EtcdCACert,
 		"experimental-initial-corrupt-check": "true",
 		"experimental-watch-progress-notify-interval": "5s",
+		"experimental-compact-hash-check-enabled":     "true",
```

Comment:
should we add a machineconfig validation such that we check if etcd is set to < 3.5.0 since it's overridable?

Reply:
let's go ahead with this.